src/java/edu/stanford/muse/index/Indexer.java

```diff
@@ -2129,6 +2139,47 @@ String getContents(org.apache.lucene.document.Document doc, boolean originalCont
         return contents;
     }
 
+	String getHeaders(edu.stanford.muse.index.Document d)
+	{
+		org.apache.lucene.document.Document doc;
+		try {
+			doc = getDoc(d);
+		} catch (IOException e) {
+			log.warn("Unable to obtain document " + d.getUniqueId() + " from index");
+			e.printStackTrace();
+			return null;
+		}
+
+		return getHeaders(doc);
+	}
+
+	String getHeaders(org.apache.lucene.document.Document doc) {
+        String headers;
+        try {
+			headers = doc.get("headers_original");
+        } catch (Exception e) {
+            log.warn("Exception " + e + " trying to read field 'headers_original': " + Util.ellipsize(Util.stackTrace(e), 350));
+			Util.print_exception("Exception Trying to read field headers_original",e,log);
+            headers = null;
+        }
+
+
+        if (headers == null) { // fall back to 'names' (or in public mode)
```

Comment:
If the header field is not present in the lucene doc then why do we want to fall back to returning named entities?